packages/@lwc/engine/src/framework/renderer.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+export type HostNode = any;
+export type HostElement = any;
+
+export interface Renderer<N = HostNode, E = HostElement> {
+    syntheticShadow: boolean;
+    insert(node: N, parent: E, anchor: N | null): void;
+    remove(node: N, parent: E): void;
```

Comment:
May be your intention was to move away from the DOM API naming convention. If so, I understand and you can ignore this suggestion. Otherwise, I would be explicit about the function's purpose and keep it close the DOM APIs because they are familiar.
```suggestion
    insertBefore(node: N, parent: E, anchor: N | null): void;
    removeChild(node: N, parent: E): void;
```

Reply:
Yes, we want to abstract away the DOM API so we agreed to keep it as is.